### PR TITLE
feat: printout compose_id when using -latest image pointer for Openstack

### PR DESCRIPTION
**Pull request by dav-pascual**
In order to know what image is the "latest" tag pointing to in compose name in Openstack,
meta_compose_id and meta_compose_url fields from image meta information are
printed out in the log and added to ansible inventory output.

Implemented in a way that allows extension for other providers with extra custom fields in the future,
or images that don't have those fields at all.

### CODE REVIEW

**Comment by dav-pascual** [comment]
As I stated, It's only implemented for Openstack images for now, and these values are optional since there are images that don't have meta_compose_id and meta_compose_url fields like Windows images. 

Compose Id and compose url are being printed out in mrack log, and mrack database has been extended to output them to ansible inventory file.

The extension of mrack database, consist of a dict in the host, which accepts such extra values. This allow us to use other values from other providers in the future.

File: tests/unit/test_ansible_inventory.py [hunk 1]
```diff
@@ -118,3 +129,64 @@ def test_invalid_layouts(self, layout, db, metadata):
         with pytest.raises(ConfigError) as excinfo:
             ans_inv.create_inventory()
         assert "dictionary" in str(excinfo.value)
+
+    @pytest.mark.parametrize(
```
Comment (pvoborni) [inline tests/unit/test_ansible_inventory.py:133]: I don't think we need to parametrize this test on layout. Layout defines have groups are nested in each other this is not related to the tested feature - host attributes.

File: tests/unit/test_ansible_inventory.py [hunk 2]
```diff
@@ -118,3 +129,64 @@ def test_invalid_layouts(self, layout, db, metadata):
...
+        first_host = inventory["all"]["hosts"][
+            metadata["domains"][0]["hosts"][0]["name"]
+        ]
+
+        assert (
+            "meta_compose_url" in first_host
+        ), "Host must have 'meta_compose_url' field"
+        assert "meta_compose_id" in first_host, "Host must have 'meta_compose_id' field"
+
+    @pytest.mark.parametrize(
```
Comment (pvoborni) [inline tests/unit/test_ansible_inventory.py:160]: Same comment as with the other parametrization.

File: tests/unit/test_ansible_inventory.py [hunk 3]
```diff
@@ -118,3 +129,64 @@ def test_invalid_layouts(self, layout, db, metadata):
...
+            False,
+        ],
+    )
+    def test_meta_extra(self, layout, db_meta_extra, metadata):
+
+        config = provisioning_config(layout)
+        ans_inv = AnsibleInventoryOutput(config, db_meta_extra, metadata)
+        inventory = ans_inv.create_inventory()
+
+        first_host = inventory["all"]["hosts"][
```
Comment (pvoborni) [inline tests/unit/test_ansible_inventory.py:151]: think you can also use e.g.
```python
first_host = inventory["all"]["hosts"].values()[0]
```
which seems easier/more readable (untested though)

File: src/mrack/host.py [hunk 4]
```diff
@@ -51,6 +51,7 @@ def host_from_json(host_data):
         host_data["username"],
         host_data["password"],
         host_data["error"],
+        host_data["meta_extra"],
```
Comment (pvoborni) [inline src/mrack/host.py:54]: This should use `host_data.get("meta_extra")` for the cases where e.g. `mrack list` was called with DB created by previous version of mrack. 

Now it causes e.g.:
```
  File "/home/pvoborni/.local/lib/python3.10/site-packages/mrack/dbdrivers/file.py", line 37, in __init__
    self.load()
  File "/home/pvoborni/.local/lib/python3.10/site-packages/mrack/dbdrivers/file.py", line 51, in load
    host = host_from_json(raw_host)
  File "/home/pvoborni/.local/lib/python3.10/site-packages/mrack/host.py", line 54, in host_from_json
    host_data["meta_extra"],
```

**Review by pvoborni — CHANGES_REQUESTED** [review]
Hi, thanks for the patch. It mostly looks good. Have few smaller comments.  

I especially appreciate the comprehensive commit message and PR description.

File: tests/unit/test_ansible_inventory.py (hunk 3 above)
Comment (dav-pascual) [reply tests/unit/test_ansible_inventory.py]: I also have observed that inventory output doesn't output host in the same order as in metadata.

**Review by Tiboris — APPROVED** [review]
thank you for the patch @dav-pascual  LGTM from my pov

File: tests/unit/test_ansible_inventory.py (hunk 3 above)
Comment (pvoborni) [reply tests/unit/test_ansible_inventory.py]: It probably doesn't matter much.  All the host should get the attr - in current mock data preparation.  Anyway, this looks also fine.

**Comment by dav-pascual** [comment]
Rebased with Black pre-commit hook patch
